### PR TITLE
improved cli

### PR DESCRIPTION
**Pull request by zeroXbrock**
supercedes #25. Upgrading matchmaker was necessary to fix the dependency tree.

- adds oclif-based CLI (will eventually replace the yarn scripts)
- adds matchmaker client
- improves gas params for sending bundles

TODO:
- [x] ship peerDependency patch to matchmaker-ts
- [x] set package.json in core to track the npm release
- [x] add new instructions in the README
- [x] resolve merge conflicts

---

- set default gasLimit for backrun tx
- add spam script
- add spam to package.json
- fix help message for spam
- remove unnecessary log
- decouple spam tx logic from controller loop
- log exceptions
- fix typo in import; breaks on linux but not mac :|
- follow base fee, improve DX
- upgrade matchmaker-ts
- remove erroneous comment
- allow 'mintIfNeeded' to fail w/o crashing the program
- use matchmaker-ts 0.5
- make build target es2016
- move library into core/
- add cli skeleton
- use 'export default' syntax in core lib export
- make cli play nice w/ other libs
- upgrade TS, use local matchmaker (temporarily), add eventsource types to devDeps
- don't check env when using mev-flood as a library
- add working init script to cli
- add helpers for dryness
- break out core spam functionality into lib
- add spam script to new cli
- load & save deployment files, fix provider bug in swap lib, simplify cli args

### CODE REVIEW

File: cli/README.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+FLAGS
+  -n, --nested-commands  Include all nested commands in the output.
+
+DESCRIPTION
+  Display help for mevflood.
+```
+
+_See code: [@oclif/plugin-help](https://github.com/oclif/plugin-help/blob/v5.2.9/src/commands/help.ts)_
+
+## `mevflood plugins`
```
Comment (metachris) [inline cli/README.md:103]: Is mevflood becoming it's own operating system now?

Does it really need a plugin system?

Comment (metachris) [reply cli/README.md:103]: Not necessarily opposed, but it's usually best to keep things simple, unless actually needed :)

**Comment by metachris** [comment]
This PR is pretty big. I think generally a bit smaller and more incremental PRs would be better.

The PR description doesn't really tell me what on the CLI has improved 🤔

File: cli/README.md (hunk 1 above)
Comment (zeroXbrock) [reply cli/README.md:103]: that's just something that oclif auto-generated, we don't use it

**Comment by zeroXbrock** [comment]
> This PR is pretty big. I think generally a bit smaller and more incremental PRs would be better.
> 
> The PR description doesn't really tell me what on the CLI has improved 🤔

yeah, I agree. This started from a spam script, and then branched to gas improvements, and then the matchmaker addition was branched from that. This is definitely not the status quo I'd like to keep in terms of PR maintenance, but in this case it's a necessary evil.

**Pull request merged**